xonsh/main.py

```diff
@@ -21,10 +26,24 @@ def main(argv=None):
 
     shell = Shell()
 
-    if args.command is None:
-        shell.cmdloop()
-    else:
+    if args.command is not None:
+        # run a single command and exit
         shell.default(args.command)
+    elif args.file is not None:
+        # run a script contained in a file
+        if os.path.isfile(args.file):
+            with open(args.file) as f:
```

Comment:
Also, I would read in the file to a local variable, then call exec outside of the file block.